app/scripts/lib/rpc-method-middleware/createMethodMiddleware.js

```diff
@@ -13,6 +13,15 @@ const handlerMap = allHandlers.reduce((map, handler) => {
   return map;
 }, new Map());
 
+const expectedHookNames = Array.from(
+  new Set(
+    Array.from(handlerMap.values()).reduce(
+      (acc, { hookNames = {} }) => acc.concat(Object.keys(hookNames)),
+      [],
+    ),
+  ).values(),
+);
```

Comment:
Nit: Seems a bit silly to take an array of handlers, insert them into a map according to the method names they handle, then take it back out of a back again and deduplicate handlers, when we could use the original handlers array instead

Would something like this be simpler, and equivalent?
```suggestion
const expectedHookNames = Array.from(
  new Set(
    allHandlers.map(({ hookNames }) => Object.keys(hookNames)).flat()
  ).values(),
);
```